src/Adyen/Service.php

```diff
@@ -43,11 +43,51 @@ public function getClient()
         return $this->client;
     }
 
+    /**
+     * @throws AdyenException
+     */
+    protected function requestHttp($url, $method = 'get', array $bodyParams = null, $requestOptions = null)
+    {
+        // check if rest api method has a value
+        if (!$method) {
+            $msg = 'The REST API method is empty';
+            $this->getClient()->getLogger()->error($msg);
+            throw new AdyenException($msg);
+        }
+        // check if rest api method has a value
+        if (!$url) {
+            $msg = 'The REST API endpoint is empty';
+            $this->getClient()->getLogger()->error($msg);
+            throw new AdyenException($msg);
+        }
+        $curlClient = $this->getClient()->getHttpClient();
+        return $curlClient->requestHttpRest($this, $url, $bodyParams, $method, $requestOptions);
+    }
+
     /**
      * @return bool
      */
     public function requiresApiKey()
     {
         return $this->requiresApiKey;
     }
+
+    /**
+     * @param string $url
+     * @return string
+     */
+    protected function createBaseUrl(string $url): string
+    {
+        $config = $this->getClient()->getConfig();
+        if ($this->getClient()->getConfig()->get('enableLive')) {
+            // replace test in string with live
+            $url = str_replace('test', 'live', $url);
+
+            // add live url prefix if needed
+            if (str_contains($url, "checkout") || str_contains($url, "pal")) {
+                $url = substr_replace($url, $config->get('prefix') . '-', 8, 0);
```

Comment:
This line contains some magic constants like "0" and "8". What about using a placeholder/variable in the url?

Reply:
Is that not a bit backwards since we still have to put that placeholder there somehow with a function like that? I can just add some documentation about how it works though.